Add colors to groups to improve readability

tdb-alcorn opened the pull request:
In a trace with many signals, differentiating groups of related traces
with color can help the eye quickly locate the signals of interest. This
change adds color to each detected group of signals, cycling through
the seven available colors in an order that ensures that adjacent colors
contrast strongly.

tdb-alcorn committed:
Add colors to groups to improve readability

In a trace with many signals, differentiating groups of related traces
with color can help the eye quickly locate the signals of interest. This
change adds color to each detected group of signals, cycling through
the seven available colors in an order that ensures that adjacent colors
contrast strongly.

seldridge commented:
:wave: @tdb-alcorn! 

This looks good (I'll test it out locally). Would you be able to re-file this PR on https://github.com/IBM/hdl-tools? I migrated this into a broader project at some point and dramatically improved performance of how tree merging works.

(I should probably archive this repository...)

tdb-alcorn committed:
Add display-vcd bash script for quicker workflow

tdb-alcorn committed:
Fix script behaviour on MacOS

On MacOS, gtkwave must be run with the system `open` command (see the
note in the README for a link to a resource explaining the details).
Unfortunately, this prevents piping stdout, so instead gtkwave must be
run with the `-O` flag to direct its output to a file. However, this
captures all of gtkwaves logging output as well, corrupting the GTKW
file we are writing to. We use awk along with start and end markers to
account for this.

This commit also adds functionality in that we now ignore any signals
whose name begins with `_`. This plays nicely with Chisel where all
the temporary signals generated begin with `_`. In a large design the
temporary signals can greatly outnumber that actual signals of interest,
so ignoring them is crucial.
